src/components/UploadForm.js

```diff
@@ -0,0 +1,93 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import Typography from '@material-ui/core/Typography';
+import TextField from '@material-ui/core/TextField';
+import Button from '@material-ui/core/Button';
+import { withStyles } from '@material-ui/core/styles';
+
+const styles = {
+  paragraph: {
+    marginTop: 50,
+    marginBottom: 10,
+  },
+  space: {
+    display: 'block',
+  },
+  button: {
+    marginTop: 20,
+  },
+};
+
+class UploadForm extends Component {
+  render () {
+    const { file, public_key, metadata, onInputChange, handleSubmit, classes } = this.props;
+
+    return (
+      <div>
+        <form onSubmit={handleSubmit}>
+          <TextField
+            id='public_key'
+            label='Public Key'
+            required
+            className={classes.space}
+            value={public_key.replace(/\s/g, '')}
+            onChange={onInputChange}
+            margin='normal'
+            fullWidth
+          />
+
+          <Typography variant='body1' className={classes.paragraph}>
+            <b>JSON file data</b> to upload. Here is some dummy data you can user, edit or replace.
+          </Typography>
+
+          <TextField
+            id='file'
+            label='file'
+            required
+            className={classes.space}
+            value={file}
+            onChange={onInputChange}
+            margin='normal'
+            multiline
+            rows='25'
+            fullWidth
+          />
+
+          <Typography variant='body1' className={classes.paragraph}>
+            <b>Metadata</b> should be text that people will use to find your data. What will be useful to query later?
```

Comment:
We're very strongly encouraging metadata to be [stringified JSON now](https://github.com/ConsenSys/linnia-resources/blob/master/METADATA.md). Would be nice to update the copy/add a check to ensure that the metadata is parseable JSON.

Reply:
Done, included the sample metadata and a link to the guidelines.